impress-org/givewp
# Onboarding Wizard selections now persist on change.

Resolves #4978

## Description



- Settings are persisted via the `onboarding/settings/{setting}` endpoint.
- Specific settings can define a static endpoint for custom handling, ie `onboarding/settings/currency`.
  - Changes to the currency also update the related currency configuration settings.
- Introduces a `SettingsRepository`, bound to the container, for getting, setting, and persisting `give_settings`.
  - The `SettingsRepository` provides a declarative interface for updating settings. This abstracts away the specifics and decouples the implementation for re-usability and testing. Namely, the `SettingsRepository` can be mocked for unit tests (which do not integrate with the database).

## Affects



- Updates the `country` and `state` keys in `app/store/reducer.js` to match the resulting setting being updated. This reduces the complexity of the endpoint handler which no longer needs to convert the setting names.

## Pre-review Checklist




## Testing Instructions



The selections made in the Onboarding Wizard now persist to the database via the REST API.

This includes, primarily, the second step in the Wizard: Location.

Updating the `base_country` should persist and update the State select.
Updating the `base_state` should persist.
Updating the `currency` should persist and additionally set the currency configuration options.

Note: While the `cause_type`, `features`, and `addons` settings are also persisted, these will soon be modified and are not of concern in this change.
